### ISSUE TITLE
jwt allows weak HMAC key to be configured

### ISSUE THREAD

**Issue by stephengaudet** [ecadlabs/signatory#356]
found on jwt-auth branch. 

steps to reproduce: 
1.configure JWT with a secret whose length is less than 32 characters

expected:
signatory fails to start, with a message to stderr explaining the minimum 32 character constraint for JWT secret

actual: 
signatory allows secret of any length

background: (taken from auth0 jwt handbook v0_14_1)
```
8.1.3 Weak HMAC Keys
HMAC algorithms rely on a shared secret to produce and verify signatures. Some people assume that shared secrets are similar to passwords, and in a sense, they are: they should be kept secret. However, that is where the similarities end. For passwords, although the length is an important property, the minimum required length is relatively small compared to other types of secrets. This is a consequence of the hashing algorithms that are used to store passwords (along with a salt) that prevent brute force attacks in reasonable timeframes.
109
On the other hand, HMAC shared secrets, as used by JWTs, are optimized for speed. This allows many sign/verify operations to be performed efficiently but make brute force attacks easier8. So, the length of the shared secret for HS256/384/512 is of the utmost importance. In fact, JSON Web Algorithms9 defines the minimum key length to be equal to the size in bits of the hash function used along with the HMAC algorithm:
“A key of the same size as the hash output (for instance, 256 bits for”HS256“) or larger MUST be used with this algorithm.” - JSON Web Algorithms (RFC 7518), 3.2 HMAC with SHA-2 Functions10
In other words, many passwords that could be used in other contexts are simply not good enough for use with HMAC-signed JWTs. 256-bits equals 32 ASCII characters, so if you are using something human readable, consider that number to be the minimum number of characters to include in the secret. Another good option is to switch to RS256 or other public-key algorithms, which are much more robust and flexible. This is not simply a hypothetical attack, it has been shown that brute force attacks for HS256 are simple enough to perform11 if the shared secret is too short.
```

**Comment by AbineshECAD** [comment]
Fixed in this commit: [add jwt per pkh integration tests](https://github.com/ecadlabs/signatory/pull/367/commits/f66ad21c4e183f72baa0a77689c5f992e021aa56)

**Comment by stephengaudet** [comment]
thank you
